core/src/test/java/org/apache/iceberg/expressions/TestExpressionParser.java

```diff
@@ -0,0 +1,265 @@
+            "    },\n" +
+            "    \"right-operand\" : {\n" +
+            "      \"type\" : \"unbounded-predicate\",\n" +
+            "      \"operation\" : \"eq\",\n" +
+            "      \"term\" : {\n" +
+            "        \"type\" : \"named-reference\",\n" +
+            "        \"value\" : \"Column2-Name\"\n" +
+            "      },\n" +
+            "      \"literals\" : [ {\n" +
+            "        \"type\" : \"string\",\n" +
+            "        \"value\" : \"Test\"\n" +
+            "      } ]\n" +
+            "    }\n" +
+            "  },\n" +
+            "  \"right-operand\" : {\n" +
+            "    \"type\" : \"unbounded-predicate\",\n" +
+            "    \"operation\" : \"is_nan\",\n" +
+            "    \"term\" : {\n" +
+            "      \"type\" : \"named-reference\",\n" +
+            "      \"value\" : \"Column3-Name\"\n" +
+            "    }\n" +
+            "  }\n" +
+            "}";
+
+    UnboundPredicate inPredicate = new UnboundPredicate(
+            Expression.Operation.IN,
+            new NamedReference("Column1-Name"),
+            Literal.of(50));
+
+    UnboundPredicate eqPredicate = new UnboundPredicate(
+            Expression.Operation.EQ,
+            new NamedReference("Column2-Name"),
+            Literal.of("Test"));
+
+    UnboundPredicate isNanPredicate = new UnboundPredicate(
+            Expression.Operation.IS_NAN,
+            new NamedReference("Column3-Name"));
+
+    And andExpression = new And(inPredicate, eqPredicate);
+    Or orNestedExpression = new Or(andExpression, isNanPredicate);
+    String actual = ExpressionParser.toJson(orNestedExpression, true);
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void aboveMaxAndBelowMinTest() {
+    String expected = "{\n" +
+            "  \"type\" : \"and\",\n" +
+            "  \"left-operand\" : {\n" +
+            "    \"type\" : \"unbounded-predicate\",\n" +
+            "    \"operation\" : \"lt\",\n" +
+            "    \"term\" : {\n" +
+            "      \"type\" : \"named-reference\",\n" +
+            "      \"value\" : \"Column1-Name\"\n" +
+            "    },\n" +
+            "    \"literals\" : [ {\n" +
+            "      \"type\" : \"above-max\"\n" +
+            "    } ]\n" +
+            "  },\n" +
+            "  \"right-operand\" : {\n" +
+            "    \"type\" : \"unbounded-predicate\",\n" +
+            "    \"operation\" : \"gt_eq\",\n" +
+            "    \"term\" : {\n" +
+            "      \"type\" : \"named-reference\",\n" +
+            "      \"value\" : \"Column2-Name\"\n" +
+            "    },\n" +
+            "    \"literals\" : [ {\n" +
+            "      \"type\" : \"below-min\"\n" +
+            "    } ]\n" +
+            "  }\n" +
+            "}";
+
+    UnboundPredicate aboveMaxPredicate = new UnboundPredicate(
+            Expression.Operation.LT,
+            new NamedReference("Column1-Name"),
+            new Literals.AboveMax());
```

Comment:
The JSON format should not handle `aboveMax` or `belowMin`. Those are only produced during binding and are immediately converted into `Expressions.alwaysTrue` or `Expressions.alwaysFalse`. It is not possible for a user to construct a predicate with either one, which is why the constructors are private and the `Literals.aboveMax()` method is package-private.